src/App.js

```diff
@@ -1,18 +1,72 @@
 import React, { Component } from 'react';
 import logo from './logo.svg';
 import './App.css';
+import { Cake } from './components/cake-card';
+import { Error } from './components/error';
+import { APIRequest } from './components/api-request';
+import { Button } from './components/button';
+import { SearchBar } from './components/search-bar';
+
+const APP_ID = '647a3750'; 
+const APP_KEY = '89aca8b5714d6d7b8972e00b3278b090	'; 
+const BASE_ENDPOINT = `https://api.edamam.com/search?app_id=${APP_ID}&app_key=${APP_KEY}`;
 
 class App extends Component {
+  constructor(){
+    super();
+    this.state = {
+      error : null,
+      cakes : [],
+      searchText : ""
+    }
+  }
+
+  onFulfilled = (data) =>{
+    this.setState({cakes:data.hits})
+  }
+  onRejected = (error) => {
+    console.log(error)
+    this.setState({error:error})
+  }
+
+  componentDidMount(){
+    APIRequest(BASE_ENDPOINT,"&q=cake","GET")
+    .then(res => res.json(), this.onRejected)
+    .then(this.onFulfilled)
+    .catch(this.onRejected);
```

Comment:
I notice here you're being explicit about your error handling, and then you're also handle _all_ cases with the `catch()`.

i would say, that since you're using `this.onRejected` to handle both the explicit one on line 34 and the catch all on line 36, you might as well leave off the one on line 34

```
    APIRequest(BASE_ENDPOINT,"&q=cake","GET")
      .then(res => res.json())
      .then(this.onFulfilled)
      .catch(this.onRejected);
```